addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/handler/ValloxSerialHandler.java

```diff
@@ -0,0 +1,397 @@
+package org.openhab.binding.vallox.handler;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.vallox.ValloxBindingConstants;
+import org.openhab.binding.vallox.internal.serial.StatusChangeListener;
+import org.openhab.binding.vallox.internal.serial.Telegram;
+import org.openhab.binding.vallox.internal.serial.ValloxProperty;
+import org.openhab.binding.vallox.internal.serial.ValloxSerialInterface;
+import org.openhab.binding.vallox.internal.serial.ValloxStore;
+import org.openhab.binding.vallox.internal.serial.ValueChangeListener;
+import org.openhab.binding.vallox.internal.serial.Variable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ValloxSerialHandler} handles all commands of the
+ * ESH runtime and interacts with the serial interface of the Vallox
+ * venting unit.
+ *
+ * For the interaction with the ESH runtime it implements {@link ThingHandler}.
+ * For getting informed about events of the vallox unit, it implements
+ * {@link ValueChangeListener} and {@StatusChangeListener}.
+ *
+ * When initializing a Thing for this binding, it opens a new connection
+ * to the vallox unit (currently via a TCP socket to a RS485 gateway),
+ * starts a thread for listening for telegrams and a second thread for
+ * a heartbeat signal. The latter is used to reliably discover when
+ * the connection is lost in order to try to reconnect.
+ *
+ * @author Hauke Fuhrmann - Initial contribution
+ *
+ */
+public class ValloxSerialHandler extends BaseThingHandler
+        implements ThingHandler, ValueChangeListener, StatusChangeListener {
+
+    private Logger logger = LoggerFactory.getLogger(ValloxSerialHandler.class);
+    private ValloxSerialInterface vallox;
+
+    public ValloxSerialHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void dispose() {
+        logger.info("dispose()");
+        if (vallox != null) {
+            vallox.stopListening();
+            vallox.getValueListener().remove(this);
+            vallox.close();
+            vallox = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("initialize()");
```

Comment:
Please change the logging level to `debug`. `trace` can be used for even more details, where necessary.